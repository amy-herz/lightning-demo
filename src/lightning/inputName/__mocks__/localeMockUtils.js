// eslint-disable-next-line lwc/no-aura-libs
import { utils as localeUtils } from 'lightning:IntlLibrary';

export function setLocaleTagMock(locale) {
    Object.assign(localeUtils, {
        getLocaleTag: jest.fn(() => locale),
    });
}
