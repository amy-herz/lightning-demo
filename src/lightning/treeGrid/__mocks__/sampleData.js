/**
 * Default settings values
 */
export const KEYFIELD = 'name';

/**
 * Columns definition
 * :: valid basic version
 */
export const COLUMNS_DEFINITION_BASIC = [
    {
        type: 'text',
        fieldName: 'accountName',
        label: 'Account Name',
    },
    {
        type: 'phone',
        fieldName: 'phone',
        label: 'Phone Number',
    },
];

/**
 * Columns definition
 * :: with non-whitelisted column keys
 */
export const COLUMNS_DEFINITION_NONWHITELIST = [
    {
        type: 'text',
        fieldName: 'accountName',
        label: 'Account Name',
    },
    {
        type: 'phone',
        fieldName: 'phone',
        label: 'Phone Number',
        sortable: true,
    },
];

/**
 * Sample expanded rows
 * :: valid basic version, no invalid row IDs
 */
export const EXPANDED_ROWS_BASIC = ['584996-s7', '377526-zg'];

/**
 * Sample expanded rows including some without children content
 * :: valid basic version, no invalid row IDs
 */
export const EXPANDED_ROWS_MISSING_CHILDREN_CONTENT = [
    '584996-s7',
    '377526-zg',
    '816682-xr',
];

/**
 * Sample expanded rows
 * :: with invalid row IDs
 */
export const EXPANDED_ROWS_INVALID = [
    '584996-s7',
    '377526-zg',
    'AWEFUL-bad_iD',
    '882894-s3',
    'PiCkLeS',
    '31337-ID',
];

/**
 * Sample selected rows
 * :: valid basic version, no invalid row IDs
 */
export const SELECTED_ROWS_BASIC = ['125313-7j', '584996-s7'];

/**
 * Sample selected rows
 * :: with invalid row IDs
 */
export const SELECTED_ROWS_INVALID = [
    '584996-s7',
    '377526-zg',
    'AWEFUL-bad_iD',
    '882894-s3',
    'PiCkLeS',
    '31337-ID',
];

/**
 * Sample data
 * :: valid basic version, no missing children content
 */
export const DATA_BASIC = [
    {
        name: '125313-7j',
        accountName: 'Dach-Welch',
        phone: '995-523-7024',
    },
    {
        name: '584996-s7',
        accountName: 'Corkery-Windler',
        phone: '782-821-1550',
        _children: [
            {
                name: '747773-jw',
                accountName: 'Corkery-Abshire',
                phone: '853-178-9812',
            },
            {
                name: '377526-zg',
                accountName: 'Robel, Friesen and Flatley',
                phone: '998-407-2397',
                _children: [
                    {
                        name: '955330-wp',
                        accountName: 'Donnelly Group',
                        phone: '793-364-7272',
                    },
                    {
                        name: '343693-9x',
                        accountName: 'Kshlerin Group',
                        phone: '507-263-1845',
                    },
                ],
            },
            {
                name: '638483-y2',
                accountName: 'Bruen, Steuber and Spencer',
                phone: '885-894-0765',
            },
        ],
    },
    {
        name: '306979-mx',
        accountName: 'Spinka LLC',
        phone: '554-511-7090',
    },
    {
        name: '066195-o1',
        accountName: 'Koelpin LLC',
        phone: '475-729-2184',
        _children: [],
    },
];

/**
 * Sample data
 * :: valid basic version, with missing children content
 */
export const DATA_MISSING_CHILDREN_CONTENT = [
    {
        name: '125313-7j',
        accountName: 'Dach-Welch',
        phone: '995-523-7024',
    },
    {
        name: '584996-s7',
        accountName: 'Corkery-Windler',
        phone: '782-821-1550',
        _children: [],
    },
    {
        name: '816682-xr',
        accountName: 'Schmitt-Littel',
        phone: '738-511-9698',
        _children: [
            {
                name: '118985-mf',
                accountName: 'Hegmann-Turcotte',
                phone: '374-284-2270',
            },
            {
                name: '841476-yo',
                accountName: 'Kuhlman LLC',
                phone: '815-872-1009',
            },
        ],
    },
    {
        name: '653331-j4',
        accountName: 'Swaniawski-Hilpert',
        phone: '366-145-6134',
        _children: [
            {
                name: '605249-ei',
                accountName: 'Swaniawski, Veum and Barton',
                phone: '792-328-3002',
            },
            {
                name: '686777-5d',
                accountName: 'Lubowitz, McClure and Russel',
                phone: '234-195-6905',
            },
            {
                name: '582166-n4',
                accountName: 'Reichel-Jerde',
                phone: '420-801-2005',
                _children: [
                    {
                        name: '513683-mm',
                        accountName: 'Tromp Inc',
                        phone: '524-614-6549',
                    },
                ],
            },
        ],
    },
    {
        name: '306979-mx',
        accountName: 'Spinka LLC',
        phone: '554-511-7090',
    },
    {
        name: '066195-o1',
        accountName: 'Koelpin LLC',
        phone: '475-729-2184',
        _children: [],
    },
];
