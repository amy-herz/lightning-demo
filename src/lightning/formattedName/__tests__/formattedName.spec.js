import { createElement } from 'lwc';
import Element from 'lightning/formattedName';
import { setLocaleTagMock } from '../__mocks__/localeUtilsFormattedName';

const createFormattedNameComponent = () => {
    const element = createElement('lightning-formatted-name', {
        is: Element,
    });
    document.body.appendChild(element);
    return element;
};

describe('lightning-formatted-name', () => {
    it('default by not passing any value', () => {
        const element = createFormattedNameComponent();
        expect(element).toMatchSnapshot();
    });

    it('defaults to en_US order and long format', () => {
        const element = createFormattedNameComponent();
        element.salutation = 'Mr.';
        element.firstName = 'John';
        element.lastName = 'Doe';
        element.middleName = 'Middleton';
        element.informalName = 'Jo';
        element.suffix = 'The 3rd';
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });
    it('Name with short format', () => {
        const element = createFormattedNameComponent();
        element.salutation = 'Mr.';
        element.firstName = 'John';
        element.lastName = 'Doe';
        element.middleName = 'Middleton';
        element.informalName = 'Jo';
        element.suffix = 'The 3rd';
        element.format = 'short';
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });
    it('Name with medium format', () => {
        const element = createFormattedNameComponent();
        element.salutation = 'Mr.';
        element.firstName = 'John';
        element.lastName = 'Doe';
        element.middleName = 'Middleton';
        element.informalName = 'Jo';
        element.suffix = 'The 3rd';
        element.format = 'medium';
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('full name with ja_JP order', () => {
        setLocaleTagMock('ja_JP');
        const element = createFormattedNameComponent();
        element.salutation = 'Mr.';
        element.firstName = 'John';
        element.lastName = 'Doe';
        element.middleName = 'Middleton';
        element.informalName = 'Jo';
        element.suffix = 'The 3rd';
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('full name with only locale set to language - ja', () => {
        setLocaleTagMock('ja');
        const element = createFormattedNameComponent();
        element.salutation = 'Mr.';
        element.firstName = 'John';
        element.lastName = 'Doe';
        element.middleName = 'Middleton';
        element.informalName = 'Jo';
        element.suffix = 'The 3rd';
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });
});
