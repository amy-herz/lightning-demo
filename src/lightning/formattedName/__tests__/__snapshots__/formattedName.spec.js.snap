// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lightning-formatted-name Name with medium format 1`] = `
<lightning-formatted-name>
  #shadow-root(open)
    John Middleton Doe
</lightning-formatted-name>
`;

exports[`lightning-formatted-name Name with short format 1`] = `
<lightning-formatted-name>
  #shadow-root(open)
    John Doe
</lightning-formatted-name>
`;

exports[`lightning-formatted-name default by not passing any value 1`] = `
<lightning-formatted-name>
  #shadow-root(open)
</lightning-formatted-name>
`;

exports[`lightning-formatted-name defaults to en_US order and long format 1`] = `
<lightning-formatted-name>
  #shadow-root(open)
    Mr. John Middleton Doe The 3rd Jo
</lightning-formatted-name>
`;

exports[`lightning-formatted-name full name with ja_JP order 1`] = `
<lightning-formatted-name>
  #shadow-root(open)
    Doe Middleton John The 3rd Jo
</lightning-formatted-name>
`;

exports[`lightning-formatted-name full name with only locale set to language - ja 1`] = `
<lightning-formatted-name>
  #shadow-root(open)
    Doe Middleton John The 3rd Jo
</lightning-formatted-name>
`;
